src/challenges/VideoSearcher.tsx

```diff
@@ -5,28 +5,65 @@ import {
   InputAdornment,
   List,
 } from "@mui/material";
-import { useState } from "react";
 import SearchIcon from "@mui/icons-material/Search";
-//import FavoriteIcon from "@mui/icons-material/Favorite";
-import { data } from "../utils/mock/videos.ts";
+
+import { useState } from "react";
+//import { info } from "../utils/mock/videos.ts";
+
+//Components
 import Video from "../components/video/Video.tsx";
 
+import { useQuery } from "@tanstack/react-query";
+
 export default function VideoSearcher() {
   const [text, setText] = useState("");
 
-  const [array, newArray] = useState(data);
+  //const [array, newArray] = useState(info);
 
-  function filteredVideos() {
-    const videos = array.filter(
-      (item) =>
-        item.name.toLowerCase().includes(text.toLowerCase()) ||
-        item.autor.toLowerCase().includes(text.toLocaleLowerCase())
-    );
+  // const [arrayVideos, setArray] = useState([]);
+  let count = 0;
+
+  const URL = "https://search.imdbot.workers.dev/?q=";
+
+  const MyURL = () => {
+    const { data, isLoading } = useQuery({
```

Comment:
It is not necesary to have this logic in another Component. As mentioned, it is better to not have components declared inside another comoponent. Please, let get rid of the MyUrl function and update its logic so the VideoSearcher is the one in cjarge of that logic. Something like how the react-query documentation shows:
![image](https://github.com/derek-0000/react-starter-challenges/assets/81263819/21334880-d725-4bd6-8bd3-d824a07300d8)
As you see, there is no need to use a component, we can use data in the final return statement.